workbench-core/datasets/src/dataSetService.test.ts

```diff
@@ -297,11 +297,15 @@ describe('DataSetService', () => {
     let service: DataSetService;
 
     beforeEach(() => {
+      metaPlugin.removeDataSet = jest.fn();
       service = new DataSetService(audit, log, metaPlugin);
     });
 
     it('returns nothing when the dataset is removed', async () => {
-      await expect(service.removeDataSet(mockDataSetId)).resolves.not.toThrow();
+      await service.removeDataSet(mockDataSetId);
+
+      expect(metaPlugin.removeDataSet).toBeCalledTimes(1);
+      expect(metaPlugin.removeDataSet).toBeCalledWith(mockDataSetId);
```

Comment:
We should be testing that were getting the expected outputs with specific inputs, not implementation details. If we change how the function works at a later date, these tests will likely fail even if the expected behavior is the same

Reply:
It's debatable - [white box testing](https://en.wikipedia.org/wiki/White-box_testing) where implementation details are known vs [black box testing](https://en.wikipedia.org/wiki/Black-box_testing) where we only know input and output.